client/me/purchases/cancel-purchase/refund-information.jsx

```diff
@@ -1,33 +1,100 @@
 import config from '@automattic/calypso-config';
 import { isDomainRegistration, isDomainMapping } from '@automattic/calypso-products';
+import { Gridicon } from '@automattic/components';
 import { localizeUrl } from '@automattic/i18n-utils';
-import i18n from 'i18n-calypso';
+import { useTranslate } from 'i18n-calypso';
 import PropTypes from 'prop-types';
+import { Fragment } from 'react';
 import { connect } from 'react-redux';
 import FormCheckbox from 'calypso/components/forms/form-checkbox';
 import FormLabel from 'calypso/components/forms/form-label';
 import FormRadio from 'calypso/components/forms/form-radio';
-import {
-	getName,
-	isRefundable,
-	isSubscription,
-	isOneTimePurchase,
-	maybeWithinRefundPeriod,
-} from 'calypso/lib/purchases';
-import { CALYPSO_CONTACT, UPDATE_NAMESERVERS } from 'calypso/lib/url/support';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import { getCancellationFeatureByKey } from 'calypso/lib/plans/cancellation-features-list';
+import { getName, isRefundable, isSubscription, isOneTimePurchase } from 'calypso/lib/purchases';
+import { UPDATE_NAMESERVERS } from 'calypso/lib/url/support';
 import { getIncludedDomainPurchase } from 'calypso/state/purchases/selectors';
+import getPlanCancellationFeatures from './get-plan-cancellation-features';
+
+const featuresList = ( translate, purchase, productSlug, hasDomain ) => {
```

Comment:
This looks like a great change for readability! As above, however, this really feels like it should be a component. If you change it to `FeaturesList` and its parameters to `( { translate, purchase, productSlug, hasDomain } )` and the call site to JSX, that's all you would need!

Reply:
FeatureList is now a React component